pkg/config/config_test.go

```diff
@@ -66,6 +66,17 @@ func TestFilterProperties(t *testing.T) {
 	assert.Check(t, properties["Captains"] != nil)
 }
 
+func TestFilterPropertiesToReturnExactMatchOnlyForConfigName(t *testing.T) {
```

Comment:
Can you also add a test for two properties with groups in their names?

E.g.:
```
m["dashboard"] = make(map[string]string)
m["dashboard"]["myProperty"] = "test"
m["dashboard.devEnv"]  = make(map[string]string)
m["dashboard.devEnv"]["myProperty"] = "test for dev"

m["dashboard-availability.devEnv"] = make(map[string]string)
```

Reply:
Thanks, yeah that makes a ton of sense. I've created the test and updated the PR already.